### DISCUSSION TITLE
Is there a way to run a synthetic forward refraction model with shot positions that are outside the line of sensors/receivers?

### DISCUSSION THREAD

**Discussion by DustinRayMIZ** [gimli-org/pyGIMLi#662] [Q&A]
I am trying to run some synthetic resolution tests against a real field survey I participated in with specifically altered geometry, but I am running into the issue of shot locations vs sensor locations. Our real setup had 128 geophones with a 2 meter spacing and we did 38 total shots with an 8 meter spacing, but we did 3 shots outside the geophones on both ends of the line (as you would usually do for a fixed refraction line). 

I have been having issues trying to set up Pygimli to run a simple forward model where I can specify shots at different locations than my sensors. The documentation seems to imply that they have to be the same for the `tt.createRAData(sensors, shotdistance)` method, which is a bit concerning to me. If anyone knows if its just hard coded to work that way, or if there's some workaround for this I would greatly appreciate any help. I am relatively new to coding in general, and I am essentially using a lightly modified version of the sloping forward model example that is available on the Pygimli website.

**Comment by halbmy** [comment]
To answer the question "Is there a way...?": Yes, of course. You just have to adapt the data container.

I agree that `tt.createRAData` is a really simple and should be extended by passing two independent arrays for shots and geophones instead of a general sensor list. This is just some simple programming task, so I appreciate anyone with a solution.
